boot/bootutil/src/loader.c

```diff
@@ -37,8 +37,10 @@
 #define BOOT_LOG_LEVEL BOOT_LOG_LEVEL_INFO
 #include "bootutil/bootutil_log.h"
 
-#ifdef MCUBOOT_MYNEWT
+#if defined(MCUBOOT_MYNEWT)
```

Comment:
I like the approach, but I don't like having the mynewt OS specific stuff in here and would rather not add more for Zephyr, especially since we need to remember to include it in every file that needs it.

Can you add a `-include mcuboot_config.h` to the compiler invocations for Zephyr?

With that change, I like the approach and would have no objections.

Reply:
I wouldn't recommend using "-include".

The problem with "-include" is that it complicates the build, this complexity again causes:

Extra build script code to be needed.

IDE's and other build system SW to struggle (E.g. CLion did not support -include for a while, and CMake will remove -include flags when there are multiple of them)

Programmers can no longer manually follow #include's to find the preprocessor macros used.

To avoid OS-specific code in here one thing one could do is something like this:

```
#if defined(USE_MCUBOOT_CONFIG)
#include <mcuboot_config.h>
#endif
```

But I will use -include if the project maintainers want it.

Reply:
Interesting; I am surprised to learn this. Since Zephyr uses `-include` and `-imacros` to pull in `${AUTOCONF_H}`, I assumed there was no problem with it. Is this something that will be addressed in Zephyr eventually?

Anyway, I'm happy to defer to your experience here and use `USE_MCUBOOT_CONFIG`. If mynewt wants to switch to it later, then the OS-specific code here for that can be removed.

Reply:
> I assumed there was no problem with it. Is this something that will be addressed in Zephyr eventually?

Yes, it's on my TODO list to get rid of it.